core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -481,6 +486,23 @@ private JarBuildItem buildThinJar(CurateOutcomeBuildItem curateOutcomeBuildItem,
         }
         Map<AppArtifactKey, List<Path>> copiedArtifacts = new HashMap<>();
 
+        Path fernflowerJar = null;
+        Path decompiledOutputDir = null;
+        boolean wasDecompiledSuccessfully = true;
+        if (packageConfig.fernflower.enabled) {
+            Path jarDirectory = Paths.get(packageConfig.fernflower.jarDirectory);
+            if (!Files.exists(jarDirectory)) {
+                Files.createDirectory(jarDirectory);
+            }
+            fernflowerJar = jarDirectory.resolve(FERNFLOWER_JAR);
+            if (!Files.exists(fernflowerJar)) {
```

Comment:
You will need to deal with updates somehow. So I think you probably need to include the git hash or a signature in the filename.

Reply:
Yeah, I guess that makes sense

Reply:
Fixed